src/ConnectComponents.tsx

```diff
@@ -14,11 +14,25 @@ export const ConnectComponentsProvider = ({
   connectInstance,
   children,
 }: {
-  connectInstance: connectJs.StripeConnectInstance;
+  connectInstance:
+    | Promise<connectJs.StripeConnectInstance>
+    | connectJs.StripeConnectInstance
+    | null;
   children: any;
 }): JSX.Element => {
+  const [resolvedConnectInstance, setResolvedConnectInstance] =
+    React.useState<connectJs.StripeConnectInstance | null>(null);
+
+  React.useEffect(() => {
```

Comment:
@michaelphines-stripe this has a race condition. If someone injects a promise and then changes this to another promise, now we have 2 of these `setResolvedConnectInstance` competing for completion. If we want to resolve the race condition, we should use something like `react-query`.

Reply:
Similarly, what happens if the `connectInstance` promise fails to resolve? Are we expecting callers to handle that error, or do we do something when that fails to resolve?

Are we also going to update our docs with this updated guidance?

Note: I like the direction of this approach, it solves a few issues for initialization, but I think there is more to do to fully take it through (in terms of error handling and perf optimization) if we believe this is the recommended way of initializing ConnectJS